### PR TITLE
Settings activity app lock

### PR DESCRIPTION
**Pull request by RoboMagus**
## Summary

This PR fixes issue #2877. 
The following changes are included:

- App-lock related processing is extracted from the `WebViewActivity` and unified such that the `SettingsActivity` can use it as well.
- A blur view was added to the `SettingsActivity` when app lock is active.
- Authentication for enabling the app-lock was migrated from the `SettingsFragment` to `SettingsActivity`, as it was impossible to have 2 different authentication initiators and callback methods in the same activity.
- A 200ms grace period was added to the app-lock timeout, to allow switching between the settings and main activities.

## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#

### CODE REVIEW

File: app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt
Comment (RoboMagus) [inline app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt:266]: Confused by why the `unlocked` status is returned here. 
If not required, not having a blocking call to `isAppLocked` here would be better.

Comment (jpelgrom) [reply app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt:266]: This function should return `true` if the gesture has been consumed/handled by the app, and `false` if it isn't (in which case it'll be passed back to the webview/system to handle). If the app is currently locked, it won't respond to the gesture, so it should return `false`.

The returned value is [used here](https://github.com/home-assistant/android/blob/cabd41e6d5f64fe5218dcbf6be2d6a838f7e6885/app/src/main/java/io/homeassistant/companion/android/util/OnSwipeListener.kt#L98), and [returned to the `View.OnTouchListener` here](https://github.com/home-assistant/android/blob/master/app/src/main/java/io/homeassistant/companion/android/util/OnSwipeListener.kt#L107), which corresponds to [the return value specified in the developer documentation here](https://developer.android.com/reference/kotlin/android/view/View.OnTouchListener?hl=en#onTouch(android.view.View,%20android.view.MotionEvent)).

Comment (RoboMagus) [reply app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt:266]: Thanks for clarifying the use of the unlock state here.

I had a gut feeling that calling `isAppLocked` here _every_ motion event could bog down the system / cause slow UI response, which is why I wanted to bring that up right away. Not sure if the function call would slow the system down too much though. 

What's your thoughts on the matter? Should we cache the unlock state in the WebViewActivity, or are the `isAppLocked` function call in these places not harmfull?

Comment (jpelgrom) [reply app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt:266]: Is there any reason why it shouldn't be cached? I don't think it is possible for this value to change between `onResume` and `onPause` aside from a lock callback, is there?

Comment (RoboMagus) [reply app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt:266]: Good point, those should indeed be the only places where the state could change.

I've made changes to keep the app lock state cached.

File: app/src/main/java/io/homeassistant/companion/android/settings/SettingsActivity.kt
Comment (jpelgrom) [inline app/src/main/java/io/homeassistant/companion/android/settings/SettingsActivity.kt:78]: I'm seeing differences with the `WebViewActivity` that seem unnecessary:

 - Why make it complicated with the window decorview root instead of using `findViewById(R.id.root)`?
 - Auto updating is enabled by default according to the documentation

File: app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt
Comment (jpelgrom) [inline app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt:100]: Because `isValid` is now no longer changed, it can be changed to a `val`

File: app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt
Comment (jpelgrom) [inline app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt:472]: What is the point of returning `success` here, as it is basically the input and can also be determined outside the function?

File: app/src/main/res/layout/activity_settings.xml
Comment (jpelgrom) [inline app/src/main/res/layout/activity_settings.xml:7]: Why are you specifying a background that is different? This changes the appearance of the settings, and creates a weird interface for elements that rely on the previous value. For example, the manage sensors interface now has striping because you've changed the background to an off-white color:
image

File: app/src/main/res/layout/activity_settings.xml
Comment (jpelgrom) [inline app/src/main/res/layout/activity_settings.xml:9]: As this is now a nested element, `xmlns:android` shouldn't be added.

**Review by jpelgrom — COMMENTED** [review]
`getSessionExpireMillis()` in the integration repository is no longer used from outside the integration repository so it should switch to a `private fun` instead.

I see a lot of log statements, are all of those still useful outside development / when the app is released? For example in `SettingsActivity.onResume` I see it logging if the app lock is enabled, but this will also become obvious immediately after as the app is locked (or not) (and is also logged by the function you're calling).

For logging, try using [Kotlin string templates](https://kotlinlang.org/docs/strings.html#string-templates) to make the code cleaner, so `"someValue: " + variable` should be `"someValue: $variable"` instead (Android Studio offers a quick fix for this).

File: app/src/main/res/layout/activity_settings.xml
Comment (jpelgrom) [reply app/src/main/res/layout/activity_settings.xml:7]: After removing it the app appears to be working fine...

File: common/src/main/java/io/homeassistant/companion/android/common/data/integration/impl/IntegrationRepositoryImpl.kt
Comment (RoboMagus) [reply common/src/main/java/io/homeassistant/companion/android/common/data/integration/impl/IntegrationRepositoryImpl.kt:84]: I could not find any other applock grace period, so the value used here is quite arbitrary.
200ms does the job for activity switching, but is generally short enough to lock the app for human interaction. E.g. quick succession of switching to a different app and back again.

File: app/src/main/java/io/homeassistant/companion/android/settings/SettingsActivity.kt
Comment (RoboMagus) [reply app/src/main/java/io/homeassistant/companion/android/settings/SettingsActivity.kt:78]: When adding the blurview to the settingsActivity, the fragment content kept showing up on top of the blur. I've made the auto update explicit to ensure the blur is redrawn on content changes, though as you indicate it seems to be redundant.
The rootView with the added background finally caused the blur to show as intended.

File: app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt
Comment (RoboMagus) [reply app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt:472]: This function is passed as a callback to the authentication handling performed in the settingsActivity. `succes` indicates if the authentication flow is completely handled, and if so the callback is removed. Since the authenticator in the settingsActivity now has to fulfill both the app locking and the original functionality from the settingsFragment, the return of success is used for the bookkeeping which authentication mode should be active. (See the use of this return value in the settingsActivity)

File: common/src/main/java/io/homeassistant/companion/android/common/data/integration/impl/IntegrationRepositoryImpl.kt
Comment (jpelgrom) [reply common/src/main/java/io/homeassistant/companion/android/common/data/integration/impl/IntegrationRepositoryImpl.kt:84]: Looking at it again I can't find the 1000 I was referring to, sorry! The 200ms seems to work fine.

File: app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt
Comment (jpelgrom) [reply app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt:472]: But if the result is not `Authenticator.SUCCESS`, you would still handle the response in this function (by changing the setting to off) while the callback won't be removed. Is that correct?

File: app/src/main/res/layout/activity_settings.xml
Comment (jpelgrom) [reply app/src/main/res/layout/activity_settings.xml:7]: Tried in on another device with a lower Android version, and now the settings screen is showing the Home Assistant logo in the background due to using the launch screen background...
image

File: app/src/main/java/io/homeassistant/companion/android/settings/SettingsActivity.kt
Comment (jpelgrom) [reply app/src/main/java/io/homeassistant/companion/android/settings/SettingsActivity.kt:78]: Please see [this comment](https://github.com/home-assistant/android/pull/2926#discussion_r985075537) - with the latest code, the background doesn't appear to be necessary for the blur to work and has other unwanted side effects.

For code consistency, switching to property access (first point) is still preferred.

**Comment by jpelgrom** [comment]
Testing on an older device (though not that old, ~3y flagship level) with commit c6dae45 I'm getting an authentication prompt when opening settings. Is that supposed to happen?

<details>
<summary>Logcat output</summary>

```
2022-10-02 11:55:27.562 26954-27100 WebviewActivity         io....stant.companion.android.debug  D  External bus {"type":"config_screen/show","id":5}
2022-10-02 11:55:27.574 26954-26954 IntegrationRepository   io....stant.companion.android.debug  D  setSessionExpireMillis(): 1664704527774
2022-10-02 11:55:27.574 26954-26954 IntegrationRepository   io....stant.companion.android.debug  D  setAppActive(): false
2022-10-02 11:55:27.576 26954-26954 IntegrationRepository   io....stant.companion.android.debug  D  setSessionExpireMillis(): 1664704527776
2022-10-02 11:55:27.576 26954-26954 IntegrationRepository   io....stant.companion.android.debug  D  setAppActive(): false
2022-10-02 11:55:27.576 26954-27162 FA                      io....stant.companion.android.debug  V  Recording user engagement, ms: 44722
2022-10-02 11:55:27.577 26954-27162 FA                      io....stant.companion.android.debug  E  Invalid google_app_id. Firebase Analytics disabled. See https://goo.gl/NAOOOI. provided id: mobilesdk_app_id
2022-10-02 11:55:27.577 26954-27162 FA                      io....stant.companion.android.debug  V  Connecting to remote service
2022-10-02 11:55:27.584 26954-27162 FA                      io....stant.companion.android.debug  V  Activity paused, time: 2550996
2022-10-02 11:55:27.596 26954-26982 FA                      io....stant.companion.android.debug  V  onActivityCreated
2022-10-02 11:55:27.626 26954-26954 UrlRepository           io....stant.companion.android.debug  D  Using external URL
2022-10-02 11:55:27.626 26954-26954 UrlRepository           io....stant.companion.android.debug  D  Using external URL
2022-10-02 11:55:27.627 26954-26954 chatty                  io....stant.companion.android.debug  I  uid=10385(io.homeassistant.companion.android.debug) identical 1 line
2022-10-02 11:55:27.628 26954-26954 UrlRepository           io....stant.companion.android.debug  D  Using external URL
2022-10-02 11:55:27.664 26954-26954 IntegrationRepository   io....stant.companion.android.debug  D  isAppLocked(). LockEnabled: true, appActive: false, expireMillis: 1664704527776, currentMillis: 1664704527664
2022-10-02 11:55:27.664 26954-27162 FA                      io....stant.companion.android.debug  V  Activity resumed, time: 2551084
2022-10-02 11:55:27.665 26954-27162 FA                      io....stant.companion.android.debug  V  Connection attempt already in progress
2022-10-02 11:55:27.786 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  --> POST https://mobile-apps.home-assistant.io/api/checkRateLimits
2022-10-02 11:55:27.786 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  Content-Type: application/json; charset=UTF-8
2022-10-02 11:55:27.786 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  Content-Length: 17
2022-10-02 11:55:27.786 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  {"push_token":""}
2022-10-02 11:55:27.786 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  --> END POST (17-byte body)
2022-10-02 11:55:27.854 26954-27112 UrlRepository           io....stant.companion.android.debug  D  localUrl is: false and usesInternalSsid is: false
2022-10-02 11:55:27.857 26954-26994 WM-Processor            io....stant.companion.android.debug  I  Moving WorkSpec (1a0dd5e3-675c-4930-95d9-bdce878bec3c) to the foreground
2022-10-02 11:55:27.863 26954-26954 IntegrationRepository   io....stant.companion.android.debug  D  isAppLocked(). LockEnabled: true, appActive: false, expireMillis: 1664704527776, currentMillis: 1664704527863
2022-10-02 11:55:27.867 26954-27112 SensorWorker            io....stant.companion.android.debug  D  Updating all Sensors in foreground.
2022-10-02 11:55:27.868 26954-27112 LastUpdate              io....stant.companion.android.debug  D  Last update is SensorWorker
2022-10-02 11:55:27.872 26954-27162 FA                      io....stant.companion.android.debug  D  Connected to remote service
2022-10-02 11:55:27.873 26954-27112 UrlRepository           io....stant.companion.android.debug  D  localUrl is: false and usesInternalSsid is: false
2022-10-02 11:55:27.874 26954-27112 UrlRepository           io....stant.companion.android.debug  D  localUrl is: false and usesInternalSsid is: false
2022-10-02 11:55:27.874 26954-27162 FA                      io....stant.companion.android.debug  V  Processing queued up service tasks: 2
2022-10-02 11:55:27.877 26954-27072 okhttp.OkHttpClient     io....stant.companion.android.debug  I  --> POST /api/webhook/d1ad019a235edc7781f5ef667d61e503189f3f0ccec7a4a00d08a5b9c9c42012
2022-10-02 11:55:27.877 26954-27072 okhttp.OkHttpClient     io....stant.companion.android.debug  I  Content-Type: application/json; charset=UTF-8
2022-10-02 11:55:27.877 26954-27072 okhttp.OkHttpClient     io....stant.companion.android.debug  I  Content-Length: 21
2022-10-02 11:55:27.877 26954-27072 okhttp.OkHttpClient     io....stant.companion.android.debug  I  {"type":"get_config"}
2022-10-02 11:55:27.877 26954-27072 okhttp.OkHttpClient     io....stant.companion.android.debug  I  --> END POST (21-byte body)
2022-10-02 11:55:27.892 26954-26954 WM-SystemFgDispatcher   io....stant.companion.android.debug  I  Started foreground service Intent { act=ACTION_START_FOREGROUND cmp=io.homeassistant.companion.android.debug/androidx.work.impl.foreground.SystemForegroundService (has extras) }
2022-10-02 11:55:27.913 26954-27072 okhttp.OkHttpClient     io....stant.companion.android.debug  I  <-- 200 OK /api/webhook/d1ad019a235edc7781f5ef667d61e503189f3f0ccec7a4a00d08a5b9c9c42012 (35ms)
2022-10-02 11:55:27.913 26954-27072 okhttp.OkHttpClient     io....stant.companion.android.debug  I  Content-Type: application/json; charset=utf-8
2022-10-02 11:55:27.913 26954-27072 okhttp.OkHttpClient     io....stant.companion.android.debug  I  Content-Length: 5658
2022-10-02 11:55:27.913 26954-27072 okhttp.OkHttpClient     io....stant.companion.android.debug  I  Date: Sun, 02 Oct 2022 09:55:27 GMT
2022-10-02 11:55:27.913 26954-27072 okhttp.OkHttpClient     io....stant.companion.android.debug  I  Server: Python/3.10 aiohttp/3.8.1
2022-10-02 11:55:27.914 26954-27072 okhttp.OkHttpClient     io....stant.companion.android.debug  I  [private data]
2022-10-02 11:55:27.914 26954-27072 okhttp.OkHttpClient     io....stant.companion.android.debug  I  <-- END HTTP (5658-byte body)
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  <-- 403 https://mobile-apps.home-assistant.io/api/checkRateLimits (170ms)
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  content-type: application/json; charset=utf-8
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  etag: W/"2c-gBARTr9vp3TOJjFE+ZvQBukBp+I"
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  function-execution-id: a93s0q6d8xtf
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  server: Google Frontend
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  x-cloud-trace-context: 56c8cb750f85c054082bf1a4e254b41e;o=1
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  accept-ranges: bytes
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  date: Sun, 02 Oct 2022 09:55:27 GMT
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  x-served-by: cache-ams21047-AMS
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  x-cache: MISS
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  x-cache-hits: 0
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  x-timer: S1664704528.798795,VS0,VE137
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  vary: x-fh-requested-host, accept-encoding
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  alt-svc: h3=":443";ma=86400,h3-29=":443";ma=86400,h3-27=":443";ma=86400
2022-10-02 11:55:27.957 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  content-length: 44
2022-10-02 11:55:27.958 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  {"errorMessage":"You did not send a token!"}
2022-10-02 11:55:27.958 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  <-- END HTTP (44-byte body)
2022-10-02 11:55:27.960 26954-27092 IntegrationRepository   io....stant.companion.android.debug  E  Unable to get notification rate limits
                                                                                                    retrofit2.HttpException: HTTP 403 
                                                                                                    	at retrofit2.KotlinExtensions$await$2$2.onResponse(KotlinExtensions.kt:53)
                                                                                                    	at retrofit2.OkHttpCall$1.onResponse(OkHttpCall.java:161)
                                                                                                    	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
                                                                                                    	at java.lang.Thread.run(Thread.java:923)
2022-10-02 11:55:27.960 26954-27092 SettingsPresenter       io....stant.companion.android.debug  D  Unable to get rate limits
2022-10-02 11:55:28.071 26954-27112 UrlRepository           io....stant.companion.android.debug  D  localUrl is: false and usesInternalSsid is: false
2022-10-02 11:55:28.075 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  --> POST /api/webhook/d1ad019a235edc7781f5ef667d61e503189f3f0ccec7a4a00d08a5b9c9c42012
2022-10-02 11:55:28.075 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  Content-Type: application/json; charset=UTF-8
2022-10-02 11:55:28.075 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  Content-Length: 277
2022-10-02 11:55:28.075 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  {"type":"update_sensor_states","data":[{"unique_id":"battery_temperature","state":27.0,"type":"sensor","icon":"mdi:battery","attributes":{}},{"unique_id":"battery_power","state":1.03,"type":"sensor","icon":"mdi:battery-plus","attributes":{"current":0.235351,"voltage":4.386}}]}
2022-10-02 11:55:28.075 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  --> END POST (277-byte body)
2022-10-02 11:55:28.117 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  <-- 200 OK /api/webhook/d1ad019a235edc7781f5ef667d61e503189f3f0ccec7a4a00d08a5b9c9c42012 (41ms)
2022-10-02 11:55:28.117 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  Content-Type: application/json; charset=utf-8
2022-10-02 11:55:28.117 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  Content-Length: 78
2022-10-02 11:55:28.117 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  Date: Sun, 02 Oct 2022 09:55:28 GMT
2022-10-02 11:55:28.117 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  Server: Python/3.10 aiohttp/3.8.1
2022-10-02 11:55:28.118 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  {"battery_temperature": {"success": true}, "battery_power": {"success": true}}
2022-10-02 11:55:28.118 26954-27078 okhttp.OkHttpClient     io....stant.companion.android.debug  I  <-- END HTTP (78-byte body)
2022-10-02 11:55:28.124 26954-26986 WM-WorkerWrapper        io....stant.companion.android.debug  I  Worker result SUCCESS for Work [ id=1a0dd5e3-675c-4930-95d9-bdce878bec3c, tags={ io.homeassistant.companion.android.sensors.SensorWorker } ]
2022-10-02 11:55:28.126 26954-26954 WM-SystemFgDispatcher   io....stant.companion.android.debug  I  Stopping foreground service
```
</details>

**Comment by RoboMagus** [comment]
> Testing on an older device (though not that old, ~3y flagship level) with commit [c6dae45](https://github.com/home-assistant/android/commit/c6dae45778a91d49d5a078b872e5596032ab3e66) I'm getting an authentication prompt when opening settings. Is that supposed to happen?

Based on the logcat output I cannot say the exact scenario in which you got the authentication prompt.

The device I use and run my tests on is a Galaxy S10 running Android 12 (also not that new anymore). Based on the testing done on this device the following behavior was validated: 
When the app is unlocked (webview) and you'd navigate to the SettingsActivity, you should not be presented with an authorization prompt. However when you let the app go to standby in the SettingsActivity and the app is re-opened after the session timeout is expired, the prompt should show. Then when navigating back to the webview after unlocking the app, the app would remain unlocked and no secondary authentication is required.

But I would not rule out that the 200ms grace period could be a bit to short for older devices?..

File: app/src/main/res/layout/activity_settings.xml
Comment (RoboMagus) [reply app/src/main/res/layout/activity_settings.xml]: > After removing it the app appears to be working fine...

I've just tested the changes where no explicit background is used, and the rootview is retrieved as per your suggestion (`window.decorView.rootView`) and it indeed seems to work just fine. 
Then I'm wondering what the issue was that caused the content to be displayed on top of the blurview, but at least it now behaves as expacted.

Thanks for the feedback! ;)

File: app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt
Comment (RoboMagus) [reply app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt]: You are correct. When the authentication is canceled the callback shoul also be deleted.

File: app/src/main/java/io/homeassistant/companion/android/settings/SettingsActivity.kt
Comment (jpelgrom) [inline app/src/main/java/io/homeassistant/companion/android/settings/SettingsActivity.kt:134]: This code is duplicated in `onWindowFocusChanged`. Looking at the `WebViewActivity`, do you need to start authenticating in both functions? Over there you only blur/unblur in `onResume` and handle authenticating in `onWindowFocusChanged`.

If required in both functions, you could extract it to another function and use that in both to remove the duplicate code.

**Review by jpelgrom — COMMENTED** [review]
Tested the latest version and now it works correctly with the increased grace period. Hopefully the 1000ms works even on old and slow devices ☺️

File: app/src/main/java/io/homeassistant/companion/android/settings/SettingsActivity.kt
Comment (RoboMagus) [reply app/src/main/java/io/homeassistant/companion/android/settings/SettingsActivity.kt]: Must have pushed the changes too soon. Handling of the authentication in `onResume` and `onWindowFocusChanged` was intended to be similar to how it's done in the webViewActivity. 

After fixing this `onResume` now only blurs, and authentication is handled only in `onWindowFocusChanged`. Just like the webViewActivity does it.

**Review by JBassett — APPROVED** [review]

**Comment by RoboMagus** [comment]
As this PR is approved, can it be merged for the next beta?

**Comment by dshokouhi** [comment]
> As this PR is approved, can it be merged for the next beta?

right now we are holding off on merges as we are stuck in review with the play store for 2022.10.1 release

**Comment by RoboMagus** [comment]
Got it, thanks for clarifying.

That also explains why I could only update to 2022.10.0 after I saw the 2022.10.1 release on github.

**Comment by JBassett** [comment]
@RoboMagus Sorry but there is now a merge conflict.  Once resolved we can merge!

**Pull request merged**
